packages/uiweb/src/lib/hooks/useAccount.ts

```diff
@@ -21,7 +21,9 @@ export const useAccount = ({env}:useAccountParams) => {
     setChain({ chainId: ethers.utils.hexValue(desiredChain) });
   };
   const provider = useMemo(() => {
-    return wallet ? new ethers.providers.Web3Provider(wallet.provider, 'any') : new ethers.providers.JsonRpcProvider(CoreRPC[env]);
+    return wallet
+      ? new ethers.providers.Web3Provider(wallet.provider, 'any')
+      : new ethers.providers.JsonRpcProvider(CoreRPC(env));
```

Comment:
Add a comment here. that this needs to move to top level context.

- First we will create a new context provider which will hold this provider for the infura calls.

- After that all the chat specific things will move to top level context.
- After that all the widget specific things will move to top level context.